packages/versioning/src/versioning.ts

```diff
@@ -922,13 +921,34 @@ export function hasDifferentReturnTypeAtVersion(
   return getReturnTypeBeforeVersion(p, type, version) !== "";
 }
 
+function findVersionMapForNamespace(program: Program, ns: Namespace): VersionMap | undefined {
+  const [_, versionMap] = getVersions(program, ns);
+  if (versionMap) {
+    return versionMap;
+  }
+  for (const [_, subNs] of ns.namespaces) {
+    const versionMap = findVersionMapForNamespace(program, subNs);
+    if (versionMap) {
+      return versionMap;
+    }
+  }
+  return undefined;
+}
+
 export function getVersionForEnumMember(program: Program, member: EnumMember): Version | undefined {
-  const parentEnum = member.enum;
-  if (!parentEnum) {
-    return undefined;
+  const globalNs = program.checker.getGlobalNamespaceType();
+  for (const [_, ns] of globalNs.namespaces) {
+    const versionMap = findVersionMapForNamespace(program, ns);
+    if (versionMap) {
+      for (const ver of versionMap.getVersions()) {
+        // FIXME: Relies on name comparisons...
+        if (member.name === ver.name) {
```

Comment:
I think this essentially requires every version enum to have a unique name across an entire spec, or the results depend on the order of processing namespaces. 

Given that nearly all of our examples use an enum called 'Versions' this feels very dangerous.

It also feels like there is potential for enums in the global namespace being inadvertently overridden by enums in the global namespace from an imported library.

I am questioning the entire premise of allowing version enums to be in the global namespace.  I think it is reasonable for versioning enums to be shared across interfaces, but the enums may need to be in a namespace so they can be uniquely identified.

Reply:
I agree that this feels dangerous--it's just the only thing that works currently, I suspect due to a bug in projections. It's as though the key in the versionMap is a projected enum and its querying with an un-projected enum, or vice versa. That's why the name comparison works when the standard === comparison fails.

I think whether the enum lives in the global namespace or not is a red herring. I think we'd run into the same problem if you declared some artificial namespace to put the version enum into.